### ISSUE TITLE
Add apache and mysql

### ISSUE THREAD

**Issue by khaldoonmasud** [wodby/docker4drupal#126]
Could you also add optional services for MySQL and apache? This will also help the developers who are using MySQL and apache instead of MariaDB and nginx.

**Comment by csandanov** [comment]
We already have a ticket requesting apache https://github.com/wodby/docker4drupal/issues/85, mariadb is a [drop-in replacement](https://en.wikipedia.org/wiki/Drop-in_replacement) of mysql

**Comment by csandanov** [comment]
Closing due to no activity

**Comment by 2dareis2do** [comment]
Because mysql does not work with persistent storage on mac or pc.

https://github.com/docker-library/mariadb/issues/95

**Comment by csandanov** [comment]
Try using `wodby/mariadb:10.1-2.3.4` (10.1.22), it should work, I use it myself. Also `wodby/mariadb:10.1-2.3.5` (10.1.26) might work. It's not that critical for local environment. They've made some changes in InnoDB in newer versions that breaks it on Win/Mac https://jira.mariadb.org/browse/MDEV-16015

**Comment by 2dareis2do** [comment]
Yeah thanks.

Funny as I had tried 10.1-2.3.5 previously and it failed. Using 10.1-2.3.4 I get the following on reboot:

my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [Note] mysqld (mysqld 10.1.22-MariaDB) starting as process 1 ...
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [Note] InnoDB: Using mutexes to ref count buffer pool pages
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [Note] InnoDB: The InnoDB memory heap is disabled
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [Note] InnoDB: Compressed tables use zlib 1.2.11
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [Note] InnoDB: Using Linux native AIO
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [Note] InnoDB: Using SSE crc32 instructions
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [Note] InnoDB: Initializing buffer pool, size = 1.0G
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [Note] InnoDB: Completed initialization of buffer pool
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [ERROR] InnoDB: auto-extending data file ./ibdata1 is of a different size 0 pages (rounded down to MB) than specified in the .cnf file: initial 640 pages, max 655360 (relevant if non-zero) pages!
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [ERROR] InnoDB: Could not open or create the system tablespace. If you tried to add new data files to the system tablespace, and it failed here, you should now edit innodb_data_file_path in my.cnf back to what it was, and remove the new ibdata files InnoDB created in this failed attempt. InnoDB only wrote those files full of zeros, but did not yet use them in any way. But be careful: do not remove old data files which contain your precious data!
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [ERROR] Plugin 'InnoDB' init function returned error.
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [Note] Plugin 'FEEDBACK' is disabled.
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [ERROR] Could not open mysql.plugin table. Some plugins may be not loaded
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [ERROR] Unknown/unsupported storage engine: InnoDB
my_drupal8_project_mariadb | 2018-06-07 13:30:17 139909200059208 [ERROR] Aborting
my_drupal8_project_mariadb | 

ok removing and recreating the data directory from previous install helps to rectify the issue.

however when doing docker-compose down 
followed by docker-compose up -d 

I get the following:

my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] mysqld (mysqld 10.1.22-MariaDB) starting as process 1 ...
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] InnoDB: Using mutexes to ref count buffer pool pages
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] InnoDB: The InnoDB memory heap is disabled
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] InnoDB: Compressed tables use zlib 1.2.11
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] InnoDB: Using Linux native AIO
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] InnoDB: Using SSE crc32 instructions
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] InnoDB: Initializing buffer pool, size = 1.0G
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] InnoDB: Completed initialization of buffer pool
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] InnoDB: Highest supported file format is Barracuda.
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] InnoDB: 128 rollback segment(s) are active.
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] InnoDB: Waiting for purge to start
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.35-80.0 started; log sequence number 48496981
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139851498162920 [Note] InnoDB: Dumping buffer pool(s) not yet started
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] Plugin 'FEEDBACK' is disabled.
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [Note] Recovering after a crash using tc.log
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [ERROR] Bad magic header in tc log
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [ERROR] Crash recovery failed. Either correct the problem (if it's, for example, out of memory error) and restart, or delete tc log and start mysqld with --tc-heuristic-recover={commit|rollback}
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [ERROR] Can't init tc log
my_drupal8_project_mariadb | 2018-06-07 18:52:05 139852998847304 [ERROR] Aborting

Basically maria is not starting for said reason
